### ISSUE TITLE
HTTP file upload no longer working with Conversations v2.2.2 on Ejabberd 18.04-2

### ISSUE THREAD

**Issue by Tyler-2** [siacs/Conversations#3066]
#### General information

* **Version:** 2.2.2
* **Device:** Motorola G4 Play
* **Android Version:** LineageOs, 14.1. Android 7.1.2.
* **Server name:** Self hosted
* **Server software:** EjabberD  18.04-2
* **Installed server modules:** Many... not sure best way to dump.
* **Conversations source:** F-Droid.

#### Steps to reproduce

1. Check Server supported XEPs in Conversation.
or
1. Try to send a file to a MUC (which requires HTTP upload).

#### Expected result

HTTP file upload shows as supported.

#### Actual result

HTTP file upload showing as unsupported. Other users running an older version of Conversations are fine. I was fine on 2.2.1. Gajim is fine.

#### Debug output

The text "HTTP", nor "0363" shows in the ADP debug log. Since there's so much private chat and contact info appearing in here, I would prefer not to share publicly.

````

````

**Comment by Tyler-2** [comment]
```  mod_http_upload:
     jid_in_url: node
     access: all
     max_size: infinity
     docroot: "/var/www/xmpp.myserver.dom/xfer"
     put_url: "https://xmpp.myserver.dom:5444"
     get_url: "https://xmpp.myserver.dom/xfer"
     thumbnail: true
```

(Domain name replaced for privacy reasons)

Using mod_http_upload as a request handler for / on port 5444, with TLS. Cert is good, confirmed with browser. Get URL is through Apache server. Also tested well.

**Comment by Tyler-2** [comment]
Possible related commits:

https://github.com/siacs/Conversations/commit/be3955ba8fafbdad3508baa02ec1060f57dc8919
https://github.com/siacs/Conversations/commit/1854e37e7ac8bed8be3cbc82741792ea20087ed4
https://github.com/siacs/Conversations/commit/2d118b5a213c583bc8babbec93f0b4d5db4d4a47

Possibly broken in the changes for P1S3? (Which I'm assuming is Amazon S3 support for uploads?)

**Comment by iNPUTmice** [comment]
Please try current master

**Comment by Tyler-2** [comment]
Unfortunately I'm not well versed on Android development and don't have any kind of build environment to do that in. Just reporting the bug and patiently waiting for the magic to appear in F-droid. :-)

**Comment by iNPUTmice** [comment]
This has been fixed in 2.2.3
